oak/server/rust/oak_runtime/tests/integration_test.rs

```diff
@@ -0,0 +1,100 @@
+//
+// Copyright 2020 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+use oak_runtime::metrics;
+use std::thread::spawn;
+
+const METRICS_PORT: u16 = 9876;
+
+mod common {
+    use oak_runtime::runtime::{Handle, Runtime};
+
+    use hyper::{Client, Uri};
+    use log::info;
+    use maplit::hashmap;
+    use oak_abi::OakStatus;
+    use oak_runtime::config;
+    use std::sync::Arc;
+    use wat::parse_str;
+
+    pub fn start_runtime() -> Result<(Arc<Runtime>, Handle), OakStatus> {
+        let wat = r#"
+        (module
+            (type (;0;) (func (param i64)))
+            (func $oak_main (type 0))
+            (memory (;0;) 18)
+            (export "memory" (memory 0))
+            (export "oak_main" (func $oak_main)))
+        "#;
+        let binary = parse_str(wat).expect("Could not parse wat module.");
+
+        // Create a runtime with one node
+        let cfg = oak_runtime::application_configuration(
+            hashmap![
+                "node".to_string() => binary,
+            ],
+            "lumberjack",
+            "node",
+            "oak_main",
+        );
+
+        info!("Starting the runtime with one nodes.");
+        config::configure_and_run(cfg)
+    }
+
+    pub async fn read_metrics() -> Result<String, hyper::error::Error> {
+        info!("Reading the metrics.");
+        let client = Client::new();
+
+        let uri = format!("http://localhost:{}", &super::METRICS_PORT)
+            .parse::<Uri>()
+            .expect("Could not parse URI.");
+
+        let res = client.get(uri).await?;
+        info!("status: {}", res.status());
+
+        let buf = hyper::body::to_bytes(res).await?;
+        Ok(std::str::from_utf8(&buf[..]).unwrap().to_string())
+    }
+}
+
+fn get_int_metric_value(all_metrics: &str, metric_name: &str) -> Option<i64> {
+    let pattern = format!(r"(.*)(\b{} \b)([0-9]+)(.*)", metric_name);
+    let re = regex::Regex::new(&pattern).unwrap();
+    re.captures(&all_metrics)
+        .map(|c| c[3].parse::<i64>().unwrap())
+}
```

Comment:
I'm not sure trying to parse metrics with regexps is particularly readable / maintainable. I would suggest, either stick to making expectations on the text itself (which I think is what you had before), or use an off-the-shelf prometheus parsing library (though that seems premature now).

Also see http://go/tott/510 .

Reply:
After the last changes the response looks like this: 
```
# HELP grpc_request_duration_seconds The gRPC request latencies in seconds.
# TYPE grpc_request_duration_seconds histogram
grpc_request_duration_seconds_bucket{le="0.005"} 0
grpc_request_duration_seconds_bucket{le="0.01"} 0
grpc_request_duration_seconds_bucket{le="0.025"} 0
grpc_request_duration_seconds_bucket{le="0.05"} 0
grpc_request_duration_seconds_bucket{le="0.1"} 0
grpc_request_duration_seconds_bucket{le="0.25"} 0
grpc_request_duration_seconds_bucket{le="0.5"} 0
grpc_request_duration_seconds_bucket{le="1"} 0
grpc_request_duration_seconds_bucket{le="2.5"} 0
grpc_request_duration_seconds_bucket{le="5"} 0
grpc_request_duration_seconds_bucket{le="10"} 0
grpc_request_duration_seconds_bucket{le="+Inf"} 0
grpc_request_duration_seconds_sum 0
grpc_request_duration_seconds_count 0
# HELP grpc_requests_total Total number of gRPC requests received.
# TYPE grpc_requests_total counter
grpc_requests_total 0
# HELP grpc_response_size_bytes The gRPC response sizes in bytes.
# TYPE grpc_response_size_bytes histogram
grpc_response_size_bytes_bucket{le="0.005"} 0
grpc_response_size_bytes_bucket{le="0.01"} 0
grpc_response_size_bytes_bucket{le="0.025"} 0
grpc_response_size_bytes_bucket{le="0.05"} 0
grpc_response_size_bytes_bucket{le="0.1"} 0
grpc_response_size_bytes_bucket{le="0.25"} 0
grpc_response_size_bytes_bucket{le="0.5"} 0
grpc_response_size_bytes_bucket{le="1"} 0
grpc_response_size_bytes_bucket{le="2.5"} 0
grpc_response_size_bytes_bucket{le="5"} 0
grpc_response_size_bytes_bucket{le="10"} 0
grpc_response_size_bytes_bucket{le="+Inf"} 0
grpc_response_size_bytes_sum 0
grpc_response_size_bytes_count 0
# HELP runtime_nodes_count Number of nodes in the runtime.
# TYPE runtime_nodes_count gauge
runtime_nodes_count 1
```
It does not make sense to assert on the whole thing. Also, as David pointed out in his comment, that kind of test would be brittle. The only off-the-shelf component I found is [prometheus-scrape](https://crates.io/crates/prometheus-scrape). But that basically does not have any documentation. So, I don't want to use it. 
For now, I'll keep this test as is.